docs/upgrade.html

```diff
@@ -202,6 +202,8 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                         </li>
                         <li>The deprecated <code>sendOffsetsToTransaction(Map&lt;TopicPartition, OffsetAndMetadata&gt;, String)</code> method has been removed from the Producer API.
                         </li>
+                        <li>The <code>flush</code> method now detects potential deadlocks and prohibits its use inside a callback. This change prevents unintended blocking behavior, which was a known risk in earlier versions.
```

Comment:
This text now needs to be in an "upgrading to 4.1" section.

Reply:
I agree with @AndrewJSchofield that this PR will not be backported to the 4.0 release. Therefore, this change should not be included in the "upgrade_4_0_0" chapter. @frankvicky, could you please create a new chapter titled "upgrade_4_1_0" and move this documentation to the "upgrade_4_1_0" chapter?

Reply:
I have applied the suggestion. Here is the preview:
![Screenshot from 2025-01-07 21-43-32](https://github.com/user-attachments/assets/015a16b1-8268-4965-8aab-e6adb0c842ce)